marshmallow_jsonapi/fields.py

```diff
@@ -134,12 +134,16 @@ def get_resource_linkage(self, value):
         if self.many:
             resource_object = [{
                 'type': self.type_,
-                'id': _stringify(get_value(each, self.id_field, each))
+                'id': _stringify(self.schema.get_attribute(each, self.id_field, each)
+                                 if _MARSHMALLOW_VERSION_INFO[0] >= 3
+                                 else self.schema.get_attribute(self.id_field, each, each))
             } for each in value]
         else:
             resource_object = {
                 'type': self.type_,
-                'id': _stringify(get_value(value, self.id_field, value))
+                'id': _stringify(self.schema.get_attribute(value, self.id_field, value)
```

Comment:
There's enough duplicated logic here that it might be worth separating this into a private helper (e.g. `_get_id(object)`).